bindings/rust/s2n-tls/src/raw/config.rs

```diff
@@ -0,0 +1,180 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::raw::error::Error;
+use alloc::rc::Rc;
+use core::convert::TryInto;
+use s2n_tls_sys::*;
+use std::ffi::CString;
+
+struct Owned(*mut s2n_config);
+
+impl Default for Owned {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Owned {
+    fn new() -> Self {
+        crate::raw::init::init();
+        let config = call!(s2n_config_new()).unwrap();
```

Comment:
what is `call!` and where is it coming from? i tried searching for it in std docs but didnt see anything

Reply:
not sure what the call pattern is for this but other fn in this file return Self

Reply:
https://github.com/aws/s2n-tls/blob/7679ff79fdcc7c3e32b32ecc6641150ddced2ace/bindings/rust/s2n-tls/src/raw/error.rs#L9-L16

It's just a macro in this codebase to ensure errors are handled properly. It puts the `unsafe` keyword inside the macro body as well, which may not be the _best_ pattern, but it got really verbose writing `unsafe` everywhere, since we're calling into C.

Reply:
hmm, really havnt done unsafe programming and dont know the best practices around it but maybe we should call the macro `unsafe_call` or not use a macro at all to hide the use of unsafe.